File: Python/DSA/LinkedList/search_linked_list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


def search(head, key):
    pos = 0
    temp = head

    while temp:
        if temp.data == key:
            return pos
        temp = temp.next
        pos += 1

    return -1


# Example
head = Node(5)
head.next = Node(10)
head.next.next = Node(15)

print("Position:", search(head, 10))

File: C/DSA/LinkedList/search_linked_list.c
#include <stdio.h>
#include <stdlib.h>


struct Node{

int data;
struct Node *next;

};



int main(){

struct Node *head;
struct Node *second;


head=malloc(sizeof(struct Node));
second=malloc(sizeof(struct Node));


head->data=5;
head->next=second;


second->data=15;
second->next=NULL;



int key=15;


struct Node *temp=head;


while(temp!=NULL){

if(temp->data==key){

printf("Found");

return 0;

}

temp=temp->next;

}


printf("Not Found");


return 0;

}